src/azureFunction/azureFunctionUtils.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as fse from 'fs-extra';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as constants from '../constants/constants';
+import * as LocalizedConstants from '../constants/localizedConstants';
+// https://github.com/microsoft/vscode-azurefunctions/blob/main/src/vscode-azurefunctions.api.d.ts
+import { AzureFunctionsExtensionApi } from '../../typings/vscode-azurefunctions.api';
+// https://github.com/microsoft/vscode-azuretools/blob/main/ui/api.d.ts
+import { AzureExtensionApiProvider } from '../../typings/vscode-azuretools.api';
+
+/**
+ * Represents the settings in an Azure function project's local.settings.json file
+ */
+export interface ILocalSettingsJson {
+	IsEncrypted?: boolean;
+	Values?: { [key: string]: string };
+	Host?: { [key: string]: string };
+	ConnectionStrings?: { [key: string]: string };
+}
+
+/**
+ * copied and modified from vscode-azurefunctions extension
+ * https://github.com/microsoft/vscode-azurefunctions/blob/main/src/funcConfig/local.settings.ts
+ * @param localSettingsPath full path to local.settings.json
+ * @returns settings in local.settings.json. If no settings are found, returns default "empty" settings
+ */
+export async function getLocalSettingsJson(localSettingsPath: string): Promise<ILocalSettingsJson> {
+	if (await fse.pathExists(localSettingsPath)) {
+		const data: string = (await fse.readFile(localSettingsPath)).toString();
+		if (/[^\s]/.test(data)) {
+			try {
+				return JSON.parse(data);
+			} catch (error) {
+				throw new Error(constants.failedToParse(error.message));
```

Comment:
Should log the original error to console in case it has some other information (such as the stack trace) that could be useful